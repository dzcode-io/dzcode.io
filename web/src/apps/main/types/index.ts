export interface SidebarTreeItem {
  id: string;
  content: string;
  link?: string;
  children?: SidebarTreeItem[];
}
