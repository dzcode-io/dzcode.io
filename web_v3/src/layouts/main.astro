<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <!-- @TODO-ZM: add head tags -->
  </head>
  <body>
    <slot />
  </body>
</html>
