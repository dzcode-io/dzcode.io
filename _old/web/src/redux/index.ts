import { combineReducers, compose, configureStore } from "@reduxjs/toolkit";
import * as slices from "src/redux/reducers";

// -----------------------------------------------------------------------------
type Slices = typeof slices;
export type SlicesKey = keyof typeof slices;
type Reducers = { [K in keyof Slices]: Slices[K]["reducer"] };
type Actions = { [K in keyof Slices]: Slices[K]["actions"] };
export type ActionType = {
  [K in keyof Actions]: `${K}/${keyof Actions[K] & string}`;
}[SlicesKey];
export type State = ReturnType<typeof rootReducer>;
// -----------------------------------------------------------------------------

const reducers = (Object.keys(slices) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({ ...pV, [sliceKey]: slices[sliceKey].reducer }),
  {} as Reducers,
);
const rootReducer = combineReducers(reducers);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const createStore = () => configureStore({ reducer: rootReducer, enhancers: composeEnhancers });

let store = createStore();

type GetStoreParam = {
  cacheStore?: boolean;
};

export const getStore = ({ cacheStore = false }: GetStoreParam = {}) => {
  if (!cacheStore) {
    store = createStore();
  }
  return store;
};

// @TODO-ZM: cache this, and add a subscriber that refreshes the cache whenever new actions get dispatched
export const getState = store.getState;

export const actions = (Object.keys(slices) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({
    ...pV,
    [sliceKey]: Object.keys(slices[sliceKey].actions).reduce(
      (contextualActions, actionName) => ({
        ...contextualActions,
        [actionName]: (...args: any[]) =>
          store.dispatch((slices[sliceKey].actions as any)[actionName](...args)),
      }),
      {},
    ),
  }),
  {} as Actions,
);
