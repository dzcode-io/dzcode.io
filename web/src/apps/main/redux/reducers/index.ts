import { articles } from "./articles";
import { articlesPage } from "./articles-page";
import { combineReducers } from "redux";
import { contributePage } from "./contribute-page";
import { documentation } from "./documentation";
import { faqPage } from "./faq-page";
import { footerComponent } from "./footer-component";
import { landingPage } from "./landing-page";
import { learnPage } from "./learn-page";
import { navbarComponent } from "./navbar-component";
import { projectsPage } from "./projects-page";
import { settings } from "./settings";
import { teamPage } from "./team-page";

export const mainReducer = combineReducers({
  documentation,
  learnPage,
  articles,
  articlesPage,
  projectsPage,
  landingPage,
  settings,
  faqPage,
  navbarComponent,
  footerComponent,
  contributePage,
  teamPage,
});
