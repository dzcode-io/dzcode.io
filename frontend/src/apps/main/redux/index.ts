import { createStore, compose } from "redux";
import { applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { mainReducer } from "./reducers";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const mainStore = createStore(
  mainReducer,
  composeEnhancers(applyMiddleware(thunk)),
);
