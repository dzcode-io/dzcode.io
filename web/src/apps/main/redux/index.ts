import { applyMiddleware, compose, createStore } from "redux";
import thunk, { ThunkAction, ThunkDispatch } from "redux-thunk";

import { mainReducer } from "./reducers";

const composeEnhancers =
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/**
 * the main redux state, with all the reducers
 */
export const mainStore = createStore(mainReducer, composeEnhancers(applyMiddleware(thunk)));

/**
 * Creates a new redux state each time this function is called, this is used only for unit tests, to ensure that we have fresh state on each individual test
 */
export const createMainStore = () => {
  return createStore(mainReducer, composeEnhancers(applyMiddleware(thunk)));
};

export type StateInterface = ReturnType<typeof mainStore.getState>;

export type ActionType =
  | "UPDATE_DOCUMENTATION"
  | "UPDATE_LEARN_PAGE"
  | "UPDATE_ARTICLES"
  | "UPDATE_ARTICLES_PAGE"
  | "UPDATE_PROJECTS"
  | "UPDATE_PROJECTS_PAGE"
  | "UPDATE_LANDING_PAGE"
  | "UPDATE_SETTINGS"
  | "UPDATE_CONTRIBUTE_PAGE";

export interface Action<T> {
  type: ActionType;
  payload: Partial<T>;
}

export type ThunkResult<A = Record<string, unknown>, E = Record<string, unknown>> = ThunkAction<
  void,
  StateInterface,
  E,
  Action<A>
>;

export type Dispatch<A> = ThunkDispatch<StateInterface, Record<string, unknown>, Action<A>>;
