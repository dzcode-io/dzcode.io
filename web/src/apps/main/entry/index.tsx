import * as Sentry from "@sentry/browser";
import { App } from "./app";
import { Integrations } from "@sentry/tracing";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { getEnv } from "src/common/utils";
import { mainStore } from "src/apps/main/redux";
import { render } from "react-dom";

Sentry.init({
  dsn: "https://b42ee25f06c4416dac83de4ab8772e26@o953637.ingest.sentry.io/5902610",
  integrations: [new Integrations.BrowserTracing()],
  tracesSampleRate: 1.0,
  environment: getEnv(),
});

render(
  <Provider store={mainStore}>
    <Router>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app-container"),
);
