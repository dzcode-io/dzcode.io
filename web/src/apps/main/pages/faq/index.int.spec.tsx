import { render, screen } from "@testing-library/react";
import { FaqPage } from ".";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { createMainStore } from "src/apps/main/redux";

describe("src/pages/landing/index.tsx", () => {
  test("Render FAQ page", async () => {
    const mainStore = createMainStore();
    const faqIndex = 0;

    const { container } = render(
      <Provider store={mainStore}>
        <Router>
          <FaqPage />
        </Router>
      </Provider>,
    );

    const firstQuestionTitle = await screen.findByTestId(`faq-title-${faqIndex}`);

    expect(firstQuestionTitle.innerHTML).toBe(mainStore.getState().faqPage.faqData[faqIndex].title);
    expect(container).toMatchSnapshot();
  });
});
