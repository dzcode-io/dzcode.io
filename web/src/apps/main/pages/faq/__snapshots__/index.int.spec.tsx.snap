// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/landing/index.tsx Render FAQ page 1`] = `
<div>
  <h4
    class="MuiTypography-root makeStyles-title-1 MuiTypography-h4"
  >
    Frequently Asked Questions
  </h4>
  <div>
    <h5
      class="MuiTypography-root makeStyles-header-2 MuiTypography-h5"
      data-testid="faq-title-0"
    >
      General
    </h5>
    <div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              What exactly is DzCode i/o ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    An open-source online community of algerian developers, no organization or any political party is involved, only us dz developers.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              What are the Goals of DzCode i/o ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <ul>
                    <li>
                      Share our experiences (in form of 
                      <a
                        href="/Articles"
                      >
                        Articles
                      </a>
                      ), there are many algerian developers that accumulated lots experiences, dzcode.io gives them the chance to share it with the right audience.
                    </li>
                    <li>
                      Guide new developers to build their career (with detailed 
                      <a
                        href="/Learn"
                      >
                        documentation
                      </a>
                      ), and understand the software market, instead of wasting their time on other useless resources.
                    </li>
                    <li>
                      Fix some common algerian software problems, in form of ready to use open-source 
                      <a
                        href="/Projects"
                      >
                        softwares
                      </a>
                       (packages, libraries ...etc).
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              What benefit will i get from it ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <div>
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
                    >
                      There are couple of benefits, to name few:
                    </p>
                    <ul>
                      <li>
                        You will meet other algerian developers, experienced and juniors, where you can grow your network!
                      </li>
                      <li>
                        You'll get to experience a professional environment, which is completely different than the one you used to in school.
                      </li>
                      <li>
                        Your contribution will be shown on your Github profile, and that is a big plus in your professional career!
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              dzcode, dzcode.io or DzCode i/o ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <div>
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
                    >
                      You may get confused with the different terms, so here's the official breakdown:
                    </p>
                    <ul>
                      <li>
                        <strong>
                          DzCode i/o
                        </strong>
                         (with capital C) is the official name of the community.
                      </li>
                      <li>
                        <strong>
                          dzcode.io
                        </strong>
                         is the website of the DzCode i/o, this has been voted for 
                        <a
                          href="https://www.facebook.com/groups/dzdevs/permalink/3052880478141052/"
                        >
                          here
                        </a>
                        .
                      </li>
                      <li>
                        <strong>
                          dzcode
                        </strong>
                         is a short name, not official, but it's usually used to refer DzCode i/o.
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h5
      class="MuiTypography-root makeStyles-header-2 MuiTypography-h5"
      data-testid="faq-title-1"
    >
      Participation
    </h5>
    <div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              How to join DzCode i/o ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    We're most active in slack, it's highly recommended to 
                    <a
                      href="https://join.slack.com/t/dzcode/shared_invite/zt-ek9kscb7-m8z_~cBjX79l~uchuABPFQ"
                    >
                      join us there
                    </a>
                    .
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              I want to code, where should I start ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <div>
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
                    >
                      Go to 
                      <a
                        href="/Contribute"
                      >
                        /Contribute
                      </a>
                       page, you will see list projects with their open tasks, pick whatever you like and start coding 🔥.
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
                    >
                      <strong>
                        Note
                      </strong>
                       : Please make sure to inform the project maintainer to reserve the task for you, you can do that by commenting on the issue ticket.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              I like the idea, what are the different ways i can contribute to DzCode i/o ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <div>
                    <p
                      class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
                    >
                      Apart from checking out the 
                      <a
                        href="/Contribute"
                      >
                        /Contribute
                      </a>
                       page, you can also contribute to DzCode i/o by:
                    </p>
                    <ul>
                      <li>
                        Adding your open-source projects to dzcode.io website, you can do that 
                        <a
                          href="/Learn/About_dzcode_io/Add_Your_Project_To_dzcode_io"
                        >
                          here
                        </a>
                        .
                      </li>
                      <li>
                        Writing about your experience in form of articles, it's as easy as following the steps 
                        <a
                          href="/Learn/About_dzcode_io/Add_Your_Article_To_dzcode_io"
                        >
                          here
                        </a>
                        .
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h5
      class="MuiTypography-root makeStyles-header-2 MuiTypography-h5"
      data-testid="faq-title-2"
    >
      Articles
    </h5>
    <div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              How to write an article in dzcode.io
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    See this 
                    <a
                      href="/Learn/About_dzcode_io/Add_Your_Article_To_dzcode_io"
                    >
                      tutorial
                    </a>
                    .
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              The Articles should they be in English ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    Yes, to reach a wide range of audience we will add a feature to allow article localization and multiple language articles in the future.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              Can I write about anything ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    As long as its IT related yes you can write about anything.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h5
      class="MuiTypography-root makeStyles-header-2 MuiTypography-h5"
      data-testid="faq-title-3"
    >
      Projects
    </h5>
    <div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              How to add my open-source project to dzcode.io
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    See this 
                    <a
                      href="/Learn/About_dzcode_io/Add_Your_Project_To_dzcode_io"
                    >
                      tutorial
                    </a>
                    .
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              What makes my project eligible to be added to dzcode.io
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <span>
                    It has to solve an Algerian problem, or, to be written by an Algerian Developer.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-outlined MuiPaper-rounded"
        style="margin-bottom: -1px;"
      >
        <div
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root makeStyles-question-3 MuiTypography-body1"
            >
              Why should i add my project to dzcode.io ?
            </p>
          </div>
          <div
            aria-disabled="false"
            aria-hidden="true"
            class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root"
                >
                  <ul>
                    <li>
                      Your project will potentially get noticed by more algerian developers.
                    </li>
                    <li>
                      You will get potential contribution from other DZ developers, and eventually better shape your piece of software.
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-spacing-4"
  />
  <p
    class="MuiTypography-root makeStyles-title-1 MuiTypography-body1"
  >
    Still need help? send us an email at
     
    <a
      href="mailto:contact@dzcode.io"
    >
      contact@dzcode.io
    </a>
  </p>
</div>
`;
